core/src/test/java/org/apache/iceberg/io/TestResolvingIO.java

```diff
@@ -47,4 +63,69 @@ public void testResolvingFileIOJavaSerialization() throws IOException, ClassNotF
     FileIO roundTripSerializedFileIO = TestHelpers.roundTripSerialize(testResolvingFileIO);
     assertThat(roundTripSerializedFileIO.properties()).isEqualTo(testResolvingFileIO.properties());
   }
+
+  @Test
+  public void resolveFileIOBulkDeletion() throws IOException {
+    // configure resolving fileIO
+    ResolvingFileIO resolvingFileIO = new ResolvingFileIO();
+    Configuration hadoopConf = new Configuration();
+    resolvingFileIO.setConf(hadoopConf);
+    resolvingFileIO.initialize(
+        ImmutableMap.of("io-impl", "org.apache.iceberg.hadoop.HadoopFileIO"));
+    ResolvingFileIO spy = spy(resolvingFileIO);
```

Comment:
nit: I don't think you have to create 2 variables, you could just init and spy the first one: `ResolvingFileIO resolvingFileIO = spy(new ResolvingFileIO());`